tests/x11/yast2_lan_restart.pm

```diff
@@ -0,0 +1,311 @@
+    else {
+        send_key 'alt-n';                             # Next
+    }
+    close_yast;
+}
+
+sub select_special_device_tab {
+    my $device = shift;
+    run_yast2_lan;
+    send_key 'tab';
+    send_key 'tab';
+    send_key 'home';
+    send_key_until_needlematch "yast2_lan_device_${device}_selected", 'down';
+    send_key 'alt-i';                                 # Edit NIC
+    assert_screen 'yast2_lan_network_card_setup';
+    if ("$device" eq 'bridge') {
+        send_key 'alt-g';                             # General
+        send_key 'alt-i';                             # Bridged devices
+        assert_screen 'yast2_lan_bridged_devices';
+    }
+    elsif ("$device" eq 'bond') {
+        send_key 'alt-o';                             # Bond slaves
+        assert_screen 'yast2_lan_bond_slaves';
+    }
+    elsif ("$device" eq 'VLAN') {
+        assert_screen 'yast2_lan_VLAN';
+    }
+    wait_still_screen;
+    send_key 'alt-n';                                 # Next
+}
+
+sub del_device {
+    my $device = shift;
+    run_yast2_lan;
+    send_key 'tab';
+    send_key 'tab';
+    send_key 'home';
+    send_key_until_needlematch "yast2_lan_device_${device}_selected", 'down';
+    send_key 'alt-t';                                 # Delete NIC
+    wait_still_screen;
+    save_screenshot;
+    send_key 'alt-i';                                 # Edit NIC
+    assert_screen 'yast2_lan_network_card_setup';
+    send_key 'alt-y';                                 # Dynamic address
+    send_key 'alt-n';                                 # Next
+    close_yast;
+}
+
+sub test_2 {
+    my $self = shift;
+    diag '__________(2) Start yast2 lan -> Edit (a NIC) -> no change, don\'t switch to another tab, [Next] -> [OK]__________';
+    run_yast2_lan_edit;
+    send_key 'alt-n';                                 # Next
+    check_network;
+}
+
+sub test_5 {
+    diag '__________(5) Start yast2 lan -> Edit (a NIC) -> no change, switch to Hardware tab, [Next] -> [OK]__________';
+    run_yast2_lan_edit;
+    send_key 'alt-w';                                 # Hardware tab
+    assert_screen 'yast2_lan_hardware_tab';
+    send_key 'alt-n';                                 # Next
+    check_network;
+}
+
+sub test_6 {
+    diag '__________(6) Start yast2 lan -> Edit (a NIC) -> no change, switch to General tab, [Next] -> [OK]__________';
+    run_yast2_lan_edit;
+    send_key 'alt-g';                                 # General tab
+    assert_screen 'yast2_lan_general_tab';
+    send_key 'alt-n';                                 # Next
+    check_network;
+}
+
+sub test_7 {
+    my $dev_name = shift;
+    diag '__________(7) Start yast2 lan -> Edit (a NIC) -> switch to Hardware tab, change nic name, [Next] -> [OK] -> device name is changed__________';
+    # workaround bsc#991694
+    record_soft_failure 'bsc#991694';
+    type_string "# workaround bsc#991694\n";
+    assert_script_run "wicked ifdown --delete eth0 sta0 dyn0";
+    # workaround bsc#991694
```

Comment:
I guess you are spelling this out a bit too often. IMHO the comments are not necessary when you already have the `record_soft_failure` *and* the `type_string` speaking for itself, right?